storage/executable-store/src/lib.rs

```diff
@@ -1,595 +0,0 @@
-/// i.e. not concurrently with other methods, and in between the piecewise execution. An
-/// exception to this occurs while the cache in the Before(id) state is updated, adding and
-/// removing elements (potentially concurrently) to convert it to contain the executables
-/// at a version corresponding to after the execution of the piece. When the update is done,
-/// it is the user's responsibility to mark the state as Updated. Currently, this happens
-/// right at the end of parallel execution inside block-executor (sequential execution mode
-/// currently does not re-use the cache acrosss pieces).
-///
-/// ExecutableStore must be pruned to control the memory usage. Currently this is done after
-/// the cache is updated, from outside the block-executor (in aptos-vm). Pruning happens
-/// based on a threshold parameter, and it simply clears the store if the size exceeds
-/// the threshold. It is the caller's responsibility to properly set the status to Pruned
-/// (ensuring we will never use the ExecutableStore without pruning).
-///
-/// After pruning, the state is supposed to change to After(id), with the id of the most
-/// recent piece that was executed. To change the state from After(id) to Before(id'), the
-/// parent of piece id' (as provided by the caller) is conformed to be id, ensuring
-/// correctness. These states are currently set in execution/executor, as the information
-/// about pieces (ids and their relation to each other) is only available there.
-///
-/// The full state transition diagram is below:
-///
-/// Empty ----------> Updated(None)
-///                      |
-///                      |
-///                      ↓
-///                   Pruned(None)
-///                      |
-///                      |
-///                      ↓
-///   --------------- After(id) -------------> Before(id')
-///   |                                            |
-///   |                                            |
-///   ↓                                            |
-/// Empty ----------> Updated(Option<ID>) <---------
-///                     ...
-///                     ...
-/// Note: After transitioning to Updated from Empty, the exact state is Updated(None),
-/// while after transitioning from Before(id'), the state is Updated(Some(id')).
-/// Similar principle applies to the Pruned state. Keeping id' allows to also confirm
-/// the consecutive nature of piece ids when marking After state.
-///
-#[derive(Copy, Clone, PartialEq, Debug)]
-enum ExecutableStoreState<ID: Clone + Debug + PartialEq> {
-    // Cache is empty and ready to be used.
-    Empty,
-    // The cache is ready-to-use at a ledger state before a piece of txns with ID id.
-    // We could represent empty as Before(None), but separating for clarity.
-    Before(ID),
-    // The cache is updated / synchronized after a piece is executed.
-    Updated(Option<ID>),
-    // The cache is pruned to control memory usage.
-    Pruned(Option<ID>),
```

Comment:
this is the only thing we don't yet have in the manager, but will do when we need I guess.

Reply:
Yes, updates are now done at boundaries, and pruning as well. Once we need this at finer regularity, we can add these states there.